### ISSUE TITLE
Issue with toolbar button active/inactive state when supplying an `onChange` handler.

### ISSUE THREAD

**Issue by SnidelyWhiplash** [wxik/react-native-rich-editor#141]
There have been a few other issues related to the button state not behaving consistently, but I think they've mostly all been addressed. However, I'm still having an issue where the active button flickers (turns inactive, then active again) on every keypress.

I've narrowed this down to the fact that I have an `onChange` handler that sets state on a parent component.

MyScreen component:
```js
render () {
  <View>
    <MyEditor
      placeholder="Enter some notes..."
      onChange={(text) => {
        this.setState({
          caption: text, // This setState causes toolbar buttons to flicker.
        })
      }}
    />
  </View>
}
```

MyEditor component:
```js
constructor () {
  this._richEditor = React.createRef()
}

render () {
  <View>
    <RichEditor
      ref={this._richEditor}
      placeholder={this.props.placeholder}
      onChange={this.props.onChange}
    />

    <RichToolbar
      editor={this._richEditor}
      actions={[
        actions.setBold,
        actions.setItalic,
        actions.setStrikethrough,
        actions.insertBulletsList,
        actions.insertOrderedList,
      ]}
    />
  </View>
}
```

Any thoughts? I need to capture changes to the RichEditor in real-time and use them on a higher component. Thanks!

**Comment by SnidelyWhiplash** [comment]
For reference, similar (possibly related) issues:
#101 
#107 
#127

**Comment by johnmoon311** [comment]
Hello!

I spent some time debugging this issue on Android. I think I have found the issue causing the rerender. It looks like the "getDerivedStateFromProps" method in the RichToolbar.js was not returning the correct mappings. I had to change the method to the below to get it to work. Please let me know if you have questions or concerns! 

`    static getDerivedStateFromProps(nextProps, prevState) {
        const {actions} = nextProps;
        if (actions !== prevState.actions) {
            let {data = []} = prevState;

            return {
                actions,
                data: actions.map((action) => ({action, selected: data.some(x => {
                    return x.action === action && x.selected;
                })})),
            };
        }
        return null;
    }`

**Comment by FloMueh** [comment]
Hi, 
i had the same issue. 

I made a fork of this lib and fixed this bug and added functionality for custom fonts. You're welcome to use this fork until this lib includes the changes.

**Comment by stulip** [comment]
Fixed `1.7.0`
